### ISSUE TITLE
Incremental clutz makes optional args non-optional when types left out due to @override

### ISSUE THREAD

**Issue by rkirov** [angular/clutz#552]
When incremental clutz goes over b.js

```javascript
class C extends B {
  /** @override */
  foo(a, b, c) {...}
}
```

it emits 

```typescript
declare class C extends B {
  foo(a: any, b: any, c: any);
}
```

However, if 'B' looks like:

```javascript
class B {
  /**
    * @param {string=} a
    * @param {string=} b
    * @param {string=} c
    */
  foo(a, b, c) {...}
}
```

Clutz invariably made optional arguments, non-optional, thus breaking consumers.

In this case there is no harm in skipping emit for simple @overrides since they are picked up from the base class.

**Comment by evmar** [comment]
Is it legal to @override and change the function signature, like adding more arguments?

**Comment by rkirov** [comment]
It is legal, but then one will have actual types to figure out what is optional and what not.

Other crazy idea is to emit `a?: any, b?: any, c?: any` when types are missing.

**Comment by evmar** [comment]
I think omitting when we see @override and no types is the most correct solution.  (Also minimizes d.ts size.)  One minor problem with it is we then drop the jsdoc, but I don't care so much.

**Comment by rkirov** [comment]
This is the entry point to the method iteration:
https://github.com/angular/clutz/blob/4ccbc8c4349c66d97f599518c2fad02dc2de1e9b/src/main/java/com/google/javascript/clutz/DeclarationGenerator.java#L2153

We already skip properties for various reason, so now we should just have one more here:

https://github.com/angular/clutz/blob/4ccbc8c4349c66d97f599518c2fad02dc2de1e9b/src/main/java/com/google/javascript/clutz/DeclarationGenerator.java#L2232

**Comment by rkirov** [comment]
New things we have learned, skipping emit does not work for types coming from implemented interfaces.

https://github.com/angular/clutz/pull/618 will make the situation around this issue a bit better, by at least keeping abstractMethods as method, albeit with all any args and return types.

**Comment by evmar** [comment]
Mostly fixed in #603, and by "fixed" I mean we just further weakened the types.

**Comment by rkirov** [comment]
Closing this as this workaround so far has proven adequate.
